core/Archive.php

```diff
@@ -733,6 +738,8 @@ private function cacheArchiveIdsWithoutLaunching($plugins)
         foreach ($plugins as $plugin) {
             $doneFlag = $this->getDoneStringForPlugin($plugin, $this->params->getIdSites());
             $this->initializeArchiveIdCache($doneFlag);
+            $globalDoneFlag = Rules::getDoneFlagArchiveContainsAllPlugins($this->params->getSegment());
+            $this->initializeArchiveIdCache($globalDoneFlag);
```

Comment:
@tsteur Did you notice there is also the call to `initializeArchiveIdCache` in `prepareArchive` method here: https://github.com/piwik/piwik/blob/2.16.1/core/Archive.php#L900-L901

should we add the initializeIdCache in this method as well?